src/libexpr/primops.cc

```diff
@@ -3031,6 +3038,13 @@ static RegisterPrimOp primop_attrValues({
     .doc = R"(
       Return the values of the attributes in the set *set* in the order
       corresponding to the sorted attribute names.
+
+      # Time Complexity
+
+      - O(n) best case (attribute set already sorted)
+      - O(n log n) worst case (requires sorting), where:
```

Comment:
There's no user-facing concept of attrset sortedness, and I don't think the best case is implemented? Usually that'd be a special case in the sorting algorithm which makes the average case worse.
Furthermore, I think pinning down the best case is too ambitious and not even useful to users.